app/src/androidTest/java/com/github/h3lp3rs/h3lp/AwaitHelpActivityTest.kt

```diff
@@ -0,0 +1,267 @@
+//            val phoneButton = onView(withId(R.id.await_help_call_button))
+//
+//            //phoneButton.inRoot(RootMatchers.isFocusable()).check(matches(isDisplayed()))
+//            phoneButton.inRoot(RootMatchers.isFocusable()).perform(click())
+//
+//            // click the contact button in the popup
+//            onView(withId(R.id.contact_call_button)).inRoot(RootMatchers.isFocusable())
+//                .perform(click())
+//
+//            // The expected ambulance phone number given the location (specified by the coordinates)
+//            val number = "tel:$VALID_CONTACT_NUMBER"
+//
+//            // Checking that this emergency number is dialed
+//            intended(
+//                allOf(
+//                    hasAction(ACTION_DIAL) //,
+//                    // hasData(Uri.parse(number)) This poses a problem for Cirrus
+//                )
+//            )
+//        }
+//    }
+
+    private fun clickingOnButtonWorksAndSendsIntent(
+        ActivityName: Class<*>?,
+        id: Matcher<View>,
+        isInScrollView: Boolean
+    ) {
+        launchAndDo(false) {
+
+            if (isInScrollView) {
+                onView(id).inRoot(RootMatchers.isFocusable()).perform(click())
+            } else {
+                onView(id).inRoot(RootMatchers.isFocusable()).perform(click())
+            }
+            intended(
+                allOf(
+                    hasComponent(ActivityName!!.name)
+                )
+            )
+        }
+    }
+
+    @Test
+    fun clickingOnHeartAttackButtonWorksAndSendsIntent() {
+        clickingOnButtonWorksAndSendsIntent(
+            HeartAttackActivity::class.java,
+            withId(R.id.heart_attack_tuto_button), true)
+    }
+
+    @Test
+    fun clickingOnEpipenButtonWorksAndSendsIntent() {
+        clickingOnButtonWorksAndSendsIntent(
+            AllergyActivity::class.java,
+            withId(R.id.epipen_tuto_button), true
+        )
+    }
+
+    @Test
+    fun clickingOnAedButtonWorksAndSendsIntent() {
+        clickingOnButtonWorksAndSendsIntent(
+            AedActivity::class.java,
+            withId(R.id.aed_tuto_button), true
+        )
+    }
+
+    @Test
+    fun cancelButtonWorksAndSendsIntent() {
+        clickingOnButtonWorksAndSendsIntent(
+            MainPageActivity::class.java,
+            withId(R.id.cancel_search_button), false
+        )
+    }
+
+    @Test
+    fun getsNotifiedWhenHelpIsComing() {
+        // Forge the right intent
+        val bundle = Bundle()
+
+        bundle.putInt(EXTRA_EMERGENCY_KEY, helpId)
+        bundle.putBoolean(EXTRA_CALLED_EMERGENCIES, true)
+        bundle.putStringArrayList(EXTRA_NEEDED_MEDICATION, arrayListOf(EPIPEN))
+
+        // Setup the database accordingly
+        val emergencyDb = MockDatabase()
+
+        val emergency = EPIPEN_EMERGENCY_INFO
+
+        emergencyDb.setObject(helpId.toString(), EmergencyInformation::class.java, emergency)
+        setDatabase(EMERGENCIES, emergencyDb)
```

Comment:
Since you already set the emergencies database to use the mock database implementation in the setup, you could directly use it here : 
```suggestion
       val emergency = EPIPEN_EMERGENCY_INFO
       val emergencyDb = databaseOf(EMERGENCIES) 
       emergencyDb.setObject(helpId.toString(), EmergencyInformation::class.java, emergency)
```